app/app.js

```diff
@@ -1270,6 +1270,23 @@ YUI.add('juju-gui', function(Y) {
             );
           }, this);
         }
+        // Handle the change set token if provided in the query.
+        // The change set token identifies a collections of changes required
+        // to deploy a bundle. Those changes are assumed to be already
+        // registered in the GUI server (via a ChangeSet:SetChanges request).
+        // Doing that is usually responsibility of a separate system
+        // (most of the times, it is Juju Quickstart).
+        var querystring = this.location.search.substring(1);
+        var qs = Y.QueryString.parse(querystring);
+        var changesToken = qs.changestoken;
+        if (Y.Lang.isValue(changesToken)) {
+          // De-dupe if necessary.
+          if (Y.Lang.isArray(changesToken)) {
```

Comment:
good catch here. This is de-duping from the question string correct? Maybe a note to that effect to mark it as different than de-duping the items in the changeset itself?